main.tf

```diff
@@ -29,6 +29,16 @@ locals {
 # Key Protect Instance
 ##############################################################################
 
+data "ibm_resource_instance" "existing_kms_instance" {
+  count      = var.existing_kms_instance_guid != null ? 1 : 0
+  identifier = var.existing_kms_instance_guid
+}
+
+locals {
+  kms_public_endpoint  = var.create_key_protect_instance ? module.key_protect[0].kp_public_endpoint : data.ibm_resource_instance.existing_kms_instance[0].extensions["endpoints.public"]
```

Comment:
what happens if the instance has no public endpoint? Will this local fail?

Reply:
It seems that even if only the private endpoint is set, the public endpoint will still exist.

[Here](https://cloud.ibm.com/projects/47cbca89-dacb-4616-a8be-76e3d0445593/configurations/5c2f0e5c-f8d9-4a90-a369-4eed820dc340#outputs) is an instance with a private endpoint:

<img width="434" alt="Screenshot 2024-06-21 at 9 00 51 PM" src="https://github.com/terraform-ibm-modules/terraform-ibm-kms-all-inclusive/assets/33781996/c51b0585-eb62-4c1c-bb09-f1aa8f8bf78d">

<img width="1087" alt="Screenshot 2024-06-21 at 8 59 53 PM" src="https://github.com/terraform-ibm-modules/terraform-ibm-kms-all-inclusive/assets/33781996/124d0ae6-c06e-4358-af76-03e7fdb1760b">

And data lookup works fine:

<img width="1507" alt="Screenshot 2024-06-21 at 8 58 13 PM" src="https://github.com/terraform-ibm-modules/terraform-ibm-kms-all-inclusive/assets/33781996/65780992-d7e8-4fb3-9b5e-36f0e5c7232c">